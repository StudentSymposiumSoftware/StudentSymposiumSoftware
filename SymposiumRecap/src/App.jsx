import { HashRouter as Router, Routes, Route, Link } from 'react-router-dom';
import { useState } from 'react'
import './App.css'

import HomepageComponent from './components/Homepage';
import SearchView from './components/SearchView';
import ListView from './components/ListView';
import GridView from './components/GridView';

/*

  Mock Data has many objects of these fields:
   - Author
   - Title
   - Category
   - Major
   - Professor
   - University
   - Abstract Number

*/
const mock_data = []


function App() {
  

  return (
    <Router>
      <div id="site-container">      
        <div id="navbar">
          Navigate to our different displays here:
          
          <Link className="PageLink" to={"/"}><h3>Home Page</h3></Link>
          <Link className="PageLink" to={"/search"}><h3>Search View</h3></Link>
          <Link className="PageLink" to={"/list"}><h3>List View</h3></Link>
          <Link className="PageLink" to={"/grid"}><h3>Grid View</h3></Link>

        </div>
      
        <Routes>
          <Route path="/" element={<HomepageComponent/>} />
          <Route path="/search" element={<SearchView/>} />
          <Route path="/list" element={<ListView/>} />
          <Route path="/grid" element={<GridView/>} />
        </Routes>
      
      </div>
    </Router>
  )
}

export default App
