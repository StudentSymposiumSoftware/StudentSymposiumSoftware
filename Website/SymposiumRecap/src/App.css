#site-container{
  width: 100vw;
  height: 100%;

  min-height: 100vh;
  background-color: #003263;

  display: flex;
  justify-content: center;
  align-items: center;
  color: white;

  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  justify-content: start;
}

#navbar{
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 100vw;
  height: 8vh;
  margin-left: 5vw;
  margin-right: 5vw;
  background-color: rgba(138, 139, 145, 0.5);
}

.PageLink{
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(117, 115, 115, 0.534);
  border-radius: 10px;
  width: 10vw;
  height: 6vh;
  text-decoration: none;
  color: rgb(0, 0, 0);
  transition: all ease-in-out 0.25s;
  border: solid rgb(100, 100, 100, 0.5) 3px;
}

.PageLink:hover{
  background-color: rgba(117, 115, 115, 0.7);
  transform: translateY(-1px)
}

.PageLink:active{
  background-color: rgba(92, 88, 88, 0.7);
  transform: translateY(1px)
}