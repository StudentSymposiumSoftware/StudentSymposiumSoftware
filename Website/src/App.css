html {
  height: 100%;
}

#site-container {
  background-color: rgb(28, 46, 63);
  color: white;
  overflow: hidden;
  height: 96%;
  /* margin: 0;
  padding: 0; */
}

.PageLink{
  padding: 10px;
  border-radius: 8px;
  color: black;
  display: inline-block;
}

.PageLink:hover{
  background-color: rgba(28, 46, 63, 0.5);
  color: white;
}

.PageLink:active{
  font-weight: bold;
}

/* select {
  color: white;
}

select:focus {
  color: black;
} */