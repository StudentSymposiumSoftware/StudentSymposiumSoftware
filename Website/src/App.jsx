import { HashRouter as Router, Routes, Route, Link } from 'react-router-dom';
import './App.css'

import HomepageComponent from './components/Homepage';
import SearchView from './components/SearchView';
import ListView from './components/ListView';
import GridView from './components/GridView';

import { getMockData } from "./mockData";

function App() {
  let mockData = getMockData();

  return (
    <Router>
      <div id="site-container">      
        <div id="navbar">
          <h3>Navigate to our different displays here:</h3>
          
          <Link className="PageLink" to={"/"}><h3>Home Page</h3></Link>
          <Link className="PageLink" to={"/search"}><h3>Search View</h3></Link>
          <Link className="PageLink" to={"/list"}><h3>List View</h3></Link>
          <Link className="PageLink" to={"/grid"}><h3>Grid View</h3></Link>

        </div>
      
        <Routes>
          <Route path="/" element={<HomepageComponent/>} />
          <Route path="/search" element={<SearchView data={mockData}/>} />
          <Route path="/list" element={<ListView data={mockData}/>} />
          <Route path="/grid" element={<GridView data={mockData}/>} />
        </Routes>
      
      </div>
    </Router>
  )
}

export default App
