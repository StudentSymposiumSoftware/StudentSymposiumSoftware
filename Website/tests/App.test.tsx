import React from 'react';
import { render } from "@testing-library/react";
import { expect, it } from "vitest";

import App from '../src/App';

it('GridView renders correctly', () => {
    const tree = render(<App/>)
    expect(tree).toMatchSnapshot();
});