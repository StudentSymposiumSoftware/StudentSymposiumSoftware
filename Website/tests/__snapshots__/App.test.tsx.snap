// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GridView renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="site-container"
      >
        <div
          class="bg-white m-5 rounded-lg ml-[20px] mr-[20px] h-[100px] w-[calc(100% - 50px)] mt-[15px] flex flex-row pr-5 drop-shadow-xl"
          id="navbar"
        >
          <span
            class="inline-flex items-center align-middle mr-5 h-[100px] p-[10px]"
          >
            <img
              class="h-[80px]"
              id="navbar_logo"
              src="logo.png"
            />
          </span>
          <span
            class="inline-flex items-center align-middle mr-5 grow"
          >
            <a
              class="PageLink"
              data-discover="true"
              href="#/admin"
            >
              <h3>
                Admin
              </h3>
            </a>
          </span>
          <span
            class="inline-flex items-center justify-items-end align-middle text-right"
          />
        </div>
        <div
          class="bg-white m-5 rounded-lg ml-[20px] mr-[20px] h-[40px] w-[calc(100% - 50px)] mt-[10px] flex flex-row pr-5 drop-shadow-xl p-[2px] hidden"
          id="filter-bar"
        >
          <span
            class="pl-[10px] inline-flex items-center align-middle grow"
          >
            <input
              class="h-[36px] text-black w-[50%] focus:outline-none flex grow"
              id="search-query"
              placeholder="Search abstracts..."
            />
          </span>
          <span
            class="p-[2px] align-middle inline-flex items-center align-middle text-right"
          >
            <a
              class="PageLink"
              href="#/list"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-primary-background h-[40px] cursor-pointer css-1umw9bq-MuiSvgIcon-root"
                data-testid="ViewListRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 14h2c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m0 5h2c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1M4 9h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m5 5h11c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m0 5h11c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1M8 6v2c0 .55.45 1 1 1h11c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1"
                />
              </svg>
            </a>
            <a
              class="PageLink"
              href="#/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-primary-background h-[40px] cursor-pointer css-1umw9bq-MuiSvgIcon-root"
                data-testid="GridViewRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 11h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2m0 10h4c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2m8-16v4c0 1.1.9 2 2 2h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-4c-1.1 0-2 .9-2 2m2 16h4c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2h-4c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2"
                />
              </svg>
            </a>
          </span>
        </div>
        <div
          class="overflow-auto h-[calc(100vh-140px)] w-[100%]"
          id="main-content"
        >
          <div
            class="grid-container"
          >
            <span
              class="loader flex place-self-center col-span-5"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="site-container"
    >
      <div
        class="bg-white m-5 rounded-lg ml-[20px] mr-[20px] h-[100px] w-[calc(100% - 50px)] mt-[15px] flex flex-row pr-5 drop-shadow-xl"
        id="navbar"
      >
        <span
          class="inline-flex items-center align-middle mr-5 h-[100px] p-[10px]"
        >
          <img
            class="h-[80px]"
            id="navbar_logo"
            src="logo.png"
          />
        </span>
        <span
          class="inline-flex items-center align-middle mr-5 grow"
        >
          <a
            class="PageLink"
            data-discover="true"
            href="#/admin"
          >
            <h3>
              Admin
            </h3>
          </a>
        </span>
        <span
          class="inline-flex items-center justify-items-end align-middle text-right"
        />
      </div>
      <div
        class="bg-white m-5 rounded-lg ml-[20px] mr-[20px] h-[40px] w-[calc(100% - 50px)] mt-[10px] flex flex-row pr-5 drop-shadow-xl p-[2px] hidden"
        id="filter-bar"
      >
        <span
          class="pl-[10px] inline-flex items-center align-middle grow"
        >
          <input
            class="h-[36px] text-black w-[50%] focus:outline-none flex grow"
            id="search-query"
            placeholder="Search abstracts..."
          />
        </span>
        <span
          class="p-[2px] align-middle inline-flex items-center align-middle text-right"
        >
          <a
            class="PageLink"
            href="#/list"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-primary-background h-[40px] cursor-pointer css-1umw9bq-MuiSvgIcon-root"
              data-testid="ViewListRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 14h2c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m0 5h2c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1M4 9h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m5 5h11c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1m0 5h11c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1M8 6v2c0 .55.45 1 1 1h11c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1"
              />
            </svg>
          </a>
          <a
            class="PageLink"
            href="#/"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-primary-background h-[40px] cursor-pointer css-1umw9bq-MuiSvgIcon-root"
              data-testid="GridViewRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 11h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2m0 10h4c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2m8-16v4c0 1.1.9 2 2 2h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-4c-1.1 0-2 .9-2 2m2 16h4c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2h-4c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2"
              />
            </svg>
          </a>
        </span>
      </div>
      <div
        class="overflow-auto h-[calc(100vh-140px)] w-[100%]"
        id="main-content"
      >
        <div
          class="grid-container"
        >
          <span
            class="loader flex place-self-center col-span-5"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
